libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -0,0 +1,46 @@
+package add_host_metadata
+
+import (
+	"github.com/shirou/gopsutil/host"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+func init() {
+	processors.RegisterPlugin("add_host_metadata", newHostMetadataProcessor)
+}
+
+type addHostMetadata struct{}
+
+func newHostMetadataProcessor(_ *common.Config) (processors.Processor, error) {
+	return &addHostMetadata{}, nil
+}
+
+func (p addHostMetadata) Run(event *beat.Event) (*beat.Event, error) {
+
+	info, err := host.Info()
```

Comment:
Do you have any benchmark on this? it would make sense to move it to the constructor and cache the result, we just need to document that a hostname change won't get reflected here

Reply:
I changed it to be instantiated in the constructor. I'm pretty sure later a request for expiring the info is coming up :-)